src/components/TodosLogic.jsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-param-reassign */
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import TodosList from './TodosList';
+import InputTodo from './InputTodo';
+
+const TodosLogic = () => {
+  const [todos, setTodos] = useState([
+    {
+      id: uuidv4(),
+      title: 'Setup development environment',
+      completed: true,
+    },
+    {
+      id: uuidv4(),
+      title: 'Develop website and add content',
+      completed: false,
+    },
+    {
+      id: uuidv4(),
+      title: 'Deploy to live server',
+      completed: false,
+    },
+  ]);
+  const handleChange = (id) => {
+    setTodos((prevState) => prevState.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+      return todo;
+    }));
+  };
+  const delTodo = (id) => {
+    setTodos([
+      ...todos.filter((todo) => todo.id !== id),
+    ]);
+  };
+  const addTodoItem = (title) => {
+    const newTodo = {
+      id: uuidv4(),
+      title,
+      completed: false,
+    };
+    setTodos([...todos, newTodo]);
+  };
+  const setUpdate = (updatedTitle, id) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          todo.title = updatedTitle;
```

Comment:
The workaround to avoid disabling the 'no-param-reassign' rule is to create a copy of the object and then reassign the title property. see the example below.

```
const todoCopy = todo;
todoCopy.title = updatedTitle;
```